frontend/src/metabase/components/ToolbarButton/ToolbarButton.tsx

```diff
@@ -0,0 +1,67 @@
+import type { ButtonHTMLAttributes, Ref, MouseEvent } from "react";
+import { forwardRef } from "react";
+
+import { color } from "metabase/lib/colors";
+import type { ActionIconProps, IconName } from "metabase/ui";
+import { ActionIcon, Icon, Tooltip } from "metabase/ui";
+
+export type ToolbarButtonProps = {
+  icon?: IconName;
+  "aria-label": string;
+  tooltipLabel?: string;
+  visibleOnSmallScreen?: boolean;
+  isActive?: boolean;
+  hasBackground?: boolean;
+} & ActionIconProps &
+  ButtonHTMLAttributes<HTMLButtonElement>;
+
+export const ToolbarButton = forwardRef(function ToolbarButton(
+  {
+    icon = "unknown",
+    "aria-label": ariaLabel,
+    onClick,
+    tooltipLabel,
+    visibleOnSmallScreen = true,
+    isActive = false,
+    hasBackground = true,
+    children,
+    disabled,
+    ...actionIconProps
+  }: ToolbarButtonProps,
+  ref: Ref<HTMLButtonElement>,
+) {
+  const handleButtonClick = (e: MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    if (onClick && !disabled) {
+      onClick?.(e);
+    }
+  };
+
+  const actionButton = (
+    <ActionIcon
+      data-testid="toolbar-button"
+      data-is-active={isActive}
+      ref={ref}
+      display={{
+        base: visibleOnSmallScreen ? "flex" : "none",
+        sm: "flex",
+      }}
+      size="2rem"
+      variant="viewHeader"
+      aria-label={ariaLabel}
+      onClick={handleButtonClick}
+      bg={hasBackground ? undefined : "transparent"}
```

Comment:
There's a regression on the question sharing button as there shouldn't be a background when hovering and the spacing is also a little different from other buttons

![image](https://github.com/user-attachments/assets/9838b59c-0149-4189-8946-d7a8ab14ec78)
![image](https://github.com/user-attachments/assets/a0f887ef-856d-4124-97a5-495ab866719a)

Reply:
@iethree I think this is where the idea of a unified `ToolbarButton` breaks. The buttons on the question view are ever-so-slightly different, so we would either need to have `QuestionToolbarButton` and `DashboardToolbarButton` , OR we get permission from the designers to use the `ToolbarButton` for everything

Reply:
Well, i'm currently working on making those sharing and alerts buttons go somewhere else anyway. We should definitely get design involved, but they're generally pretty in favor of making things look the same.

https://github.com/metabase/metabase/issues/46191

Reply:
👆 that looks awesome.

Reply:
Ok, that makes sense. I'm thinking of creating a new PR just for the ToolbarButton so you can use it asap, and then I'll split the rest into reviewable chunks so we can get design involved. Let me do that in the next few

Reply:
it's not imminently blocking me. I've copied it into my PR to work off for now. don't sweat about it.